### PR TITLE
Add include support for Ruby DSL

### PR DESCRIPTION
**Pull request by wujiandong**
add ruby dsl configure to support include

### CODE REVIEW

File: lib/fluent/config/dsl.rb [hunk 1]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
         end
 
         def self.parse(source, source_path="config.rb")
+          confpath = File.dirname(File.expand_path(source_path))
+          endres = parse_include(source,confpath,[])
           Proxy.new('ROOT', nil).eval(source, source_path).to_config_element
         end
+        def self.parse_include(content,confpath=nil,attrs=[])
+          attrs.push(content)
+          require 'httpclient'
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: fluentd doesn't depend on `httpclient` gem so don't use such library in the core.

File: lib/fluent/config/dsl.rb [hunk 2]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
...
+                path = u.path
+                if path[0] != ?/ and path[-1] == ?/
+                    pattern = File.join(File.join(confpath,path),"*.rb")
+                elsif path[0] == ?/ and  FileTest::directory? path
+                    pattern = File.join(path,"*.rb")
+                elsif path[0] == ?/ and path[-1] != ?/  and FileTest::file? path
+                    pattern = path
+              end
+                files = Dir.glob(pattern.to_s)
+                if ! files.length.eql?(0) then
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: `if files.length.non_zero?`

File: lib/fluent/config/dsl.rb [hunk 3]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
...
+                  t = File.read(path)
+                  dirpath = File.dirname(path)
+                  attrs.push(t.force_encoding('utf-8'))
+                  parse_include(t,dirpath,attrs)
+                  }
+                else
+                  raise ConfigParseError, "include error #{pattern} not exists!"
+                end
+            else
+              $log.info url,"parse_include"
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: In default configuration format, does fluentd dump these logs?

File: lib/fluent/config/dsl.rb [hunk 4]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
...
+              attrs.push(out.content.force_encoding('utf-8'))
+              parse_include(out.content.force_encoding('utf-8'),nil,attrs=attrs)
+            end
+          end
+          return attrs.uniq.join("\n")
+          rescue => e
+            raise ConfigParseError, "about include #{confpath} error "
+        end
       end
-
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: Don't remove brank line between class definitions.

File: lib/fluent/config/dsl.rb [hunk 5]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
...
+            else
+              $log.info url,"parse_include"
+              out = clt.get(url)
+              attrs.push(out.content.force_encoding('utf-8'))
+              parse_include(out.content.force_encoding('utf-8'),nil,attrs=attrs)
+            end
+          end
+          return attrs.uniq.join("\n")
+          rescue => e
+            raise ConfigParseError, "about include #{confpath} error "
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: `about` is needed?

File: lib/fluent/config/dsl.rb [hunk 6]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
...
         end
+        def self.parse_include(content,confpath=nil,attrs=[])
+          attrs.push(content)
+          require 'httpclient'
+          clt = HTTPClient.new
+          res = /(include|@include).*/m.match(content).to_s
+          res.split("\n").each do |i|
+            content.gsub!(i,"")
+            i.strip!
+            if ! i.start_with?("#")
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: `unless i.start_with?("#")`

File: lib/fluent/config/dsl.rb [hunk 7]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
         end
 
         def self.parse(source, source_path="config.rb")
+          confpath = File.dirname(File.expand_path(source_path))
+          endres = parse_include(source,confpath,[])
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: Please put whitespace between each arguments like below.
`parse_include(source, confpath, [])`

Please apply to other functions.

File: lib/fluent/config/dsl.rb [hunk 8]
```diff
@@ -27,10 +28,55 @@ def self.read(path)
...
+            if ! i.start_with?("#")
+              url=i.split(%r{\s+})[1]
+              url.gsub!("\"","")
+              u = URI.parse(url)
+              if u.scheme == 'file' || u.path == url
+
+                path = u.path
+                if path[0] != ?/ and path[-1] == ?/
+                    pattern = File.join(File.join(confpath,path),"*.rb")
+                elsif path[0] == ?/ and  FileTest::directory? path
```
Comment (repeatedly) [inline lib/fluent/config/dsl.rb]: `FileTest::directory?(path)`

**Comment by repeatedly** [comment]
It breaks tests.

File: lib/fluent/config/dsl.rb (hunk 1 above)
Comment (wujiandong) [inline lib/fluent/config/dsl.rb]: HI 
it can  open-uri

**Comment by tagomoris** [comment]
There are many wrong indentations.

**Comment by repeatedly** [comment]
Related to https://github.com/fluent/fluentd/issues/360

**Comment by tagomoris** [comment]
It's done at #950.

**Pull request closed without merging**